comp/dogstatsd/server/server_worker.go

```diff
@@ -42,7 +42,7 @@ func newWorker(s *server) *worker {
 	return &worker{
 		server:  s,
 		batcher: batcher,
-		parser:  newParser(s.sharedFloat64List),
+		parser:  newParser(s.config, s.sharedFloat64List),
```

Comment:
Unrelated to your PR once again (sorry :D) but this addition IMO shows that `newParser` should be a func part of the `server` struct..

Reply:
If this is because of passing the config component deeply, we could resolve this by passing the dependent configs individually as constructor arguments to remove that dependency. 

Or did you have something else in mind?